### PR TITLE
Merge pull request #13819 from panyx0718/doc

### PR DESCRIPTION
**Pull request by panyx0718**
Explain LoD and a few other concepts

### CODE REVIEW

File: paddle/fluid/pybind/pybind.cc [hunk 1]
```diff
@@ -156,7 +156,50 @@ PYBIND11_PLUGIN(core) {
...
+    lod_tensor.lod() can retrieve the LoD information.
+
+    LoD is short for Level of Details and is usually used for varied sequence
+    length. You can skip the following comment if you don't need optional LoD.
+
+  For example:
+     A LoDTensor X can look like the example below. It contains 2 sequences.
+     The first has length 2 and the second has length 3, as described by x.lod.
+
+     The first tensor dimension 6=2+3 is calculated from LoD if it's available.
```
Comment (luotao1) [inline paddle/fluid/pybind/pybind.cc]: @panyx0718 The first tensor dimension 6=2+3? 5=2+3

Comment (panyx0718) [reply paddle/fluid/pybind/pybind.cc]: Fixed

Comment (luotao1) [reply paddle/fluid/pybind/pybind.cc]: Should fix in the develop branch as well.

File: paddle/fluid/pybind/pybind.cc [hunk 2]
```diff
@@ -156,7 +156,50 @@ PYBIND11_PLUGIN(core) {
...
+     columns, hence [5, 2].
+
+      x.lod  = [[2, 3]]
+      x.data = [[1, 2], [3, 4],  // seq 1
+                [5, 6], [7, 8], [9, 10]]  // seq 2
+      x.shape = [5, 2]
+
+      LoD can have multiple levels (for example, a paragraph can have multiple
+      sentences and a sentence can have multiple words). In the following
+      LodTensor Y, the lod_level is 2. It means there are 2 sequence, the
```
Comment (chengduoZH) [inline paddle/fluid/pybind/pybind.cc:183]: Maybe use `2 paragraphs` here is better, but not  `2 sequence`.

Comment (panyx0718) [reply paddle/fluid/pybind/pybind.cc:183]: Here, I want to describe in general (not just nlp), so I use sequence.

**Review by chengduoZH — APPROVED** [review]
Excellent

**Pull request merged**